core/controllers/suggestion_test.py

```diff
@@ -475,3 +702,218 @@ def test_accept_question_suggestion(self):
+                    'skill_id': None
+                },
+                'description': 'Add new question to skill'
+            }, csrf_token=csrf_token)
+
+        self.logout()
+
+    def test_cannot_access_suggestion_to_topic_handler(self):
+        self.login(self.OWNER_EMAIL)
+
+        thread_id = feedback_services.create_thread(
+            suggestion_models.TARGET_TYPE_QUESTION, self.topic_id,
+            self.author_id, 'description', '', has_suggestion=True)
+
+        suggestion_models.GeneralSuggestionModel.create(
+            suggestion_models.SUGGESTION_TYPE_ADD_QUESTION,
+            suggestion_models.TARGET_TYPE_QUESTION, self.topic_id,
+            1, suggestion_models.STATUS_IN_REVIEW, self.author_id,
+            None, {
+                'cmd': (
+                    question_domain
+                    .CMD_CREATE_NEW_FULLY_SPECIFIED_QUESTION),
+                'question_dict': self.question_dict,
+                'skill_id': None
+            }, 'score_category', thread_id)
+
+        response = self.get_html_response(feconf.CREATOR_DASHBOARD_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        with self.swap(constants, 'ENABLE_NEW_STRUCTURE_PLAYERS', False):
+            self.put_json(
+                '%s/topic/%s/%s' % (
+                    feconf.SUGGESTION_ACTION_URL_PREFIX, self.topic_id,
+                    thread_id),
+                {}, csrf_token=csrf_token, expected_status_int=404)
+
+        self.logout()
+
+    def test_suggestion_to_topic_handler_with_invalid_target_type(self):
+        self.login(self.OWNER_EMAIL)
+
+        thread_id = feedback_services.create_thread(
+            suggestion_models.TARGET_TYPE_QUESTION, self.topic_id,
+            self.author_id, 'description', '', has_suggestion=True)
+
+        suggestion_models.GeneralSuggestionModel.create(
+            suggestion_models.SUGGESTION_TYPE_ADD_QUESTION,
+            suggestion_models.TARGET_TYPE_QUESTION, self.topic_id,
+            1, suggestion_models.STATUS_IN_REVIEW, self.author_id,
+            None, {
+                'cmd': (
+                    question_domain
+                    .CMD_CREATE_NEW_FULLY_SPECIFIED_QUESTION),
+                'question_dict': self.question_dict,
+                'skill_id': None
+            }, 'score_category', thread_id)
+
+        response = self.get_html_response(feconf.CREATOR_DASHBOARD_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        with self.swap(constants, 'ENABLE_NEW_STRUCTURE_PLAYERS', True):
+            response = self.put_json(
+                '%s/topic/%s/%s' % (
+                    feconf.SUGGESTION_ACTION_URL_PREFIX,
+                    self.topic_id, thread_id),
+                {}, csrf_token=csrf_token, expected_status_int=400)
+
+        self.assertEqual(
+            response['error'],
+            'This handler allows actions only on suggestions to topics.')
```

Comment:
This scenario should ideally throw an error I think, this controller is meant to be used to accept/reject SUGGESTION_TYPE_ADD_QUESTION suggestions. Maybe I am a bit confused, but could you clarify?

Reply:
Yes it is throwing an error that's why I asserted the response['error']. Its throwing a 400 error

Reply:
Oops a typo from me... What I meant to write was it **shouldn't** throw an error...

Reply:
From the above comment... The model should have target type as TARGET_TYPE_TOPIC and not TARGET_TYPE_QUESTION. Use the services to create the models and not the storage layer directly.

Reply:
No this handler should work for TARGET_TYPE_TOPIC not SUGGESTION_TYPE_ADD_QUESTION, that's why there is an error

Reply:
done